src/mpid/ch4/shm/posix/eager/fbox/fbox_recv.h

```diff
@@ -130,15 +130,15 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_POSIX_eager_recv_memcpy(void *dst, const voi
     MPIR_Memcpy(dst, src, size);
 }
 
-MPL_STATIC_INLINE_PREFIX void
-MPIDI_POSIX_eager_recv_commit(MPIDI_POSIX_eager_recv_transaction_t * transaction)
+MPL_STATIC_INLINE_PREFIX void MPIDI_POSIX_eager_recv_commit(void)
 {
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_FBOX_EAGER_RECV_COMMIT);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_FBOX_EAGER_RECV_COMMIT);
 
     MPIDI_POSIX_fastbox_t *fbox_in = NULL;
+    MPIDI_POSIX_eager_fbox_transport_t *transport = MPIDI_POSIX_eager_fbox_get_transport();
```

Comment:
Does this work if there's more than one fbox send/recv going on at a time?

Reply:
I may just be confused here because of the ordering of the commits. It seems like perhaps some later commits fix this?

Reply:
In case of multithreading fastbox transport should be accessed in a mutually exclusive fashion by each thread. This won't work for cases in which every thread processes fastboxes concurrently, as the transport has only one `fbox/cell` pointer per process. For such cases there should be an array of pointers. Do you think that should be the case?

Reply:
I was thinking more about having two isend/irecv calls at the same time to different processes.

Reply:
I was thinking more about having two isend/irecv calls at the same time to different processes.

Reply:
Every process consumes `fbox/cells` one after the other right? And in case of multiple recvs, before the next can be processed, the current one has to be released. Am I missing anything? For the send case everything should be as for current master.

Reply:
Only the cell that matches the receiving process is consumed. So if rank 0 is doing this:

```
MPI_Isend(..., dest=1, ...);
MPI_Isend(..., dest=2, ...);
MPI_Waitall(...);
```

I think both would happen at the same time.

Reply:
@wesbland apologies for not replying earlier. I don't think I see the problem your are trying to make here. In a single threaded case, which I think you are assuming here, the two isend will be serialized; I don't understand what you mean with "would happen at the same time".  In this case rank 0 will fill fastbox 0 on rank 1 and then fastbox 0 on rank 2. This behavior does not change compared with the previous implementation, at least as far as I can tell. If you want we can clarify offline.

Reply:
So if you have two operations outstanding (regardless of threading level), wouldn't the second operation overwrite `transport->curr_fbox`? So when would you progress the first one?

Reply:
`transport->curr_fbox` is set by the receiver at `recv_begin` time and reset at the next `recv_begin`. Currently, every `recv_begin` is immediately followed by a `recv_end` to make the fbox available again. The only problem arises for `MPI_THREAD_MULTIPLE` in which there might be concurrent `recv_begin` (this case was mentioned by my first reply).

Reply:
Ok. I think I misunderstood something here. This is fine.